src/diffusers/models/transformers/transformer_ltx.py

```diff
@@ -0,0 +1,468 @@
+
+        self.scale_shift_table = nn.Parameter(torch.randn(2, inner_dim) / inner_dim**0.5)
+        self.time_embed = AdaLayerNormSingle(inner_dim, use_additional_conditions=False)
+
+        self.caption_projection = PixArtAlphaTextProjection(in_features=caption_channels, hidden_size=inner_dim)
+
+        self.rope = LTXRotaryPosEmbed(
+            dim=inner_dim,
+            base_num_frames=20,
+            base_height=2048,
+            base_width=2048,
+            patch_size=patch_size,
+            patch_size_t=patch_size_t,
+            theta=10000.0,
+        )
+
+        self.transformer_blocks = nn.ModuleList(
+            [
+                LTXTransformerBlock(
+                    dim=inner_dim,
+                    num_attention_heads=num_attention_heads,
+                    attention_head_dim=attention_head_dim,
+                    cross_attention_dim=cross_attention_dim,
+                    qk_norm=qk_norm,
+                    activation_fn=activation_fn,
+                    attention_bias=attention_bias,
+                    attention_out_bias=attention_out_bias,
+                    eps=norm_eps,
+                    elementwise_affine=norm_elementwise_affine,
+                )
+                for _ in range(num_layers)
+            ]
+        )
+
+        self.norm_out = nn.LayerNorm(inner_dim, eps=1e-6, elementwise_affine=False)
+        self.proj_out = nn.Linear(inner_dim, out_channels)
+
+        self.gradient_checkpointing = False
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if hasattr(module, "gradient_checkpointing"):
+            module.gradient_checkpointing = value
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        encoder_hidden_states: torch.Tensor,
+        timestep: torch.LongTensor,
+        encoder_attention_mask: torch.Tensor,
+        rope_interpolation_scale: Optional[Tuple[float, float, float]] = None,
+        return_dict: bool = True,
+    ) -> torch.Tensor:
+        image_rotary_emb = self.rope(hidden_states, rope_interpolation_scale)
+
+        # convert encoder_attention_mask to a bias the same way we do for attention_mask
+        if encoder_attention_mask is not None and encoder_attention_mask.ndim == 2:
+            encoder_attention_mask = (1 - encoder_attention_mask.to(hidden_states.dtype)) * -10000.0
+            encoder_attention_mask = encoder_attention_mask.unsqueeze(1)
+
+        batch_size, num_channels, num_frames, height, width = hidden_states.shape
+        p = self.config.patch_size
+        p_t = self.config.patch_size_t
+
+        post_patch_height = height // p
+        post_patch_width = width // p
+        post_patch_num_frames = num_frames // p_t
+
+        hidden_states = hidden_states.reshape(
+            batch_size, -1, post_patch_num_frames, p_t, post_patch_height, p, post_patch_width, p
+        )
+        hidden_states = hidden_states.permute(0, 2, 4, 6, 1, 3, 5, 7).flatten(4, 7).flatten(1, 3)
```

Comment:
is there any reason we reshape from 5d -> 3d and then 3d -> 5d on every iteration?

Reply:
think it will be nice to:
1. have a `_pack_latent` and `_unpack_latent` like we did for flux
2.  maybe we can move the rotary embedding in the pipeline so we only pack/unpack once (I know we currently have some discrepancy here, so open to discussion); or we can just configure the rotary pos embeds class in pipeline, so we do not need to give the shape info each time

Reply:
Yes, I'll add the pack and unpack latent methods.

Regarding the RoPE, I think a separate layer approach is okay despite it requiring recomputation at every step. This is because we're planning to work on caching hooks that would enable the outputs of any layer to be cached and re-used. Since RoPE is an integral part of many models, there can be some opt-out code we could default to for enabling caching by default on these kinds of model specific RoPE layers. WDYT?